azure-iot-device/tests/common/pipeline/test_pipeline_events_base.py

```diff
@@ -3,13 +3,21 @@
 # Licensed under the MIT License. See License.txt in the project root for
 # license information.
 # --------------------------------------------------------------------------
-import pytest
+import sys
 from azure.iot.device.common.pipeline import pipeline_events_base
+from tests.common.pipeline import pipeline_data_object_test
 
+this_module = sys.modules[__name__]
 
-@pytest.mark.describe("PipelineEvent object")
-class TestPipelineEvent(object):
-    @pytest.mark.it("Sets name attribute on instantiation")
-    def test_default_arguments(self):
-        obj = pipeline_events_base.PipelineEvent()
-        assert obj.name is obj.__class__.__name__
+pipeline_data_object_test.add_event_test(
+    obj=pipeline_events_base.PipelineEvent,
```

Comment:
Shouldn't this be an AbstractBaseClass, and thus, not directly under test?

Reply:
Good point.  Will change.

---
In reply to: [291674656](https://github.com/Azure/azure-iot-sdk-python-preview/pull/104#discussion_r291674656) [](ancestors = 291674656)